bindings/rust/aws-lc-sys-template/src/lib.rs

```diff
@@ -1,9 +1,61 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0 OR ISC
 
-pub use bindings::*;
+// Warn to use feature native_bindings if building on a platform where prebuilt-bindings
+// aren't available
+#[cfg(all(
+    not(feature = "native_bindings"),
+    not(any(
+        all(target_os = "linux", target_arch = "x86"),
+        all(target_os = "linux", target_arch = "x86_64"),
+        all(target_os = "linux", target_arch = "aarch64"),
+        all(target_os = "macos", target_arch = "x86_64")
+    ))
+))]
+compile_error!("Prebuilt-bindings aren't available. Turn on feature native_bindings to build.");
 
-mod bindings;
+#[cfg(all(feature = "native_bindings", feature = "generate"))]
+compile_error!(
+    "Generate is only for internal usage. Only turn on feature native_bindings to build."
+);
+
+macro_rules! use_bindings {
+    ($bindings:ident) => {
+        mod $bindings;
+        pub use $bindings::*;
+    };
+}
+
+#[cfg(all(
+    not(feature = "native_bindings"),
+    target_os = "linux",
+    target_arch = "x86"
+))]
+use_bindings!(linux_x86_bindings);
+
+#[cfg(all(
+    not(feature = "native_bindings"),
+    target_os = "linux",
+    target_arch = "x86_64"
+))]
+use_bindings!(linux_x86_64_bindings);
+
+#[cfg(all(
+    not(feature = "native_bindings"),
+    target_os = "linux",
+    target_arch = "aarch64"
+))]
+use_bindings!(linux_aarch64_bindings);
+
+#[cfg(all(
+    not(feature = "native_bindings"),
+    target_os = "macos",
+    target_arch = "x86_64"
+))]
+use_bindings!(macos_x86_64_bindings);
+
+#[cfg(feature = "native_bindings")]
+use_bindings!(bindings);
```

Comment:
The [cfg_aliases](https://docs.rs/cfg_aliases/latest/cfg_aliases/) crate would allow us to simplify this.  It might help us auto-detect when bindings need to be generated so the user doesn't need to specify the feature.
```
cfg_aliases! {
    linux_x86_bindings: { all(not(feature = "bindgen"), target_os = "linux", target_arch = "x86") },
    linux_aarch64_bindings: { all(not(feature = "bindgen"), target_os = "linux", target_arch = "aarch64") },
    linux_x86_64_bindings: { all(not(feature = "bindgen"), target_os = "linux", target_arch = "x86_64") },
    macos_x86_64_bindings: { all(not(feature = "bindgen"), target_os = "macos", target_arch = "x86_64") },
    bindgen: { not(any(linux_x86_bindings, linux_aarch64_bindings, linux_x86_64_bindings, macos_x86_64_bindings) },
}

#[cfg(linux_x86_bindings)]
use_bindings!(linux_x86_bindings);

#[cfg(linux_aarch64_bindings)]
use_bindings!(linux_aarch64_bindings);

#[cfg(linux_x86_64_bindings)]
use_bindings!(linux_x86_64_bindings);

#[cfg(macos_x86_64_bindings)]
use_bindings!(macos_x86_64_bindings);

#[cfg(any(bindgen, feature = "generate_bindings")]
use_bindings!(bindings);
```